### ISSUE TITLE
RF02 incorrectly raised for pyformat placeholder templates

### ISSUE THREAD

**Issue by ibrokemypie** [quarylabs/sqruff#2019]
### What Happened
sqruff raises `references.consistent` for `%(key)s` when it shouldn't, as `key` here is a template variable not a column name. sqlfluff configured the same way does not raise this.

```
15:48:45 > sqruff lint test.sql
== [test.sql] FAIL
L:   3 | P:   5 | RF03 | Unqualified reference 'key' found in single table
                       | select. [references.consistent]
L:   3 | P:   5 | RF03 | Unqualified reference 'key' found in single table
                       | select which is inconsistent with previous references.
                       | [references.consistent]
The linter processed 1 file(s).
All Finished 📜 🎉
```

### Expected Behaviour

No warnings/errors. 

```
15:48:41 > sqlfluff lint test.sql
All Finished 📜 🎉!                                                                                                                                   
```

### How to reproduce

test.sql:
```sql
SELECT
    usr.name,
    %(key)s AS key_val
FROM auth_user AS usr;

```

### Configuration
```ini
[sqruff]
dialect = postgres
rules = all
templater = placeholder

[sqruff:templater:placeholder]
param_style = pyformat
```
